sdk/keyvault/azkeys/utils_test.go

```diff
@@ -22,15 +23,35 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-var pathToPackage = "sdk/keyvault/azkeys/testdata"
-
-const fakeKvURL = "https://fakekvurl.vault.azure.net/"
-const fakeKvMHSMURL = "https://fakekvurl.managedhsm.azure.net/"
+const (
+	fakeVaultURL = "https://fakekvurl.vault.azure.net"
+	fakeMHSMURL  = "https://fakekvurl.managedhsm.azure.net"
+)
 
-var enableHSM = true
+var (
+	enableHSM     bool
+	liveMHSMURL   string
+	liveVaultURL  string
+	pathToPackage = "sdk/keyvault/azkeys/testdata"
+)
 
 func TestMain(m *testing.M) {
-	// Initialize
+	liveVaultURL = strings.TrimSuffix(os.Getenv("AZURE_KEYVAULT_URL"), "/")
+	liveMHSMURL = strings.TrimSuffix(os.Getenv("AZURE_MANAGEDHSM_URL"), "/")
+	enableHSM = liveMHSMURL != ""
+
+	if recording.GetRecordMode() != recording.LiveMode {
+		err := recording.ResetProxy(nil)
```

Comment:
Why do we need to do this before/after the tests?  Is this just one of those "ensure the proxy is in a known state" type of thing?

Reply:
The latter. Just trying to prevent hard to debug problems when working on multiple modules.